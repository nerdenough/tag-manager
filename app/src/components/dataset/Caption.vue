<script setup lang="ts">
type Props = {
  caption: string
  remove: (caption: string) => void
}

defineProps<Props>()
</script>

<template>
  <div class="flex gap-x-2 bg-slate-800 text-lg p-2 rounded-md">
    <span class="flex-1">{{ caption }}</span>
    <div class="flex items-center">
      <button
        @click="() => remove(caption)"
        class="w-8 h-8 bg-slate-900 hover:bg-slate-700 text-slate-400 hover:text-red-300 rounded-md"
      >
        x
      </button>
    </div>
  </div>
</template>
