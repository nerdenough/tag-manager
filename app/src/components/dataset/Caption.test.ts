import { mount } from '@vue/test-utils'
import Caption from './Caption.vue'
import { describe, it, expect } from 'vitest'

describe('Caption', () => {
  it('should render with the specified caption text', () => {
    const caption = 'Some caption'
    const wrapper = mount(Caption, { props: { caption, remove: () => {} } })

    expect(wrapper.get('span').text()).toBe(caption)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should fire the remove function prop when remove button is pressed', () => {
    const caption = 'Some caption'
    const remove = (val: string) => {
      expect(val).toBe(caption)
    }
    const wrapper = mount(Caption, { props: { caption, remove } })

    wrapper.get('button').trigger('click')
  })
})
