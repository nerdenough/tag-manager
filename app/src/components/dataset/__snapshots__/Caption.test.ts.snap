// Vitest Snapshot v1

exports[`Caption > should render with the specified caption text 1`] = `
"<div class=\\"flex gap-x-2 bg-slate-800 text-lg p-2 rounded-md\\"><span class=\\"flex-1\\">Some caption</span>
  <div class=\\"flex items-center\\"><button class=\\"w-8 h-8 bg-slate-900 hover:bg-slate-700 text-slate-400 hover:text-red-300 rounded-md\\"> x </button></div>
</div>"
`;
